airflow/cli/commands/db.py

```diff
@@ -283,13 +294,24 @@ def shell(ctx):
 @cli_utils.action_cli(check_db=False)
 def check(ctx, migration_wait_timeout):
     """Runs a check command that checks if db is reachable"""
+    from airflow.utils import db as db_utils
+
     console = Console()
     console.print(f"Waiting for {migration_wait_timeout}s")
     db_utils.check_migrations(timeout=migration_wait_timeout)
 
 
 # lazily imported by CLI parser for `help` command
-all_tables = sorted(config_dict)
+# Create a custom class that emulates a callable since click validates
+# non-callable and make __str__ to return output for lazy processing.
+class _CleanTableDefault:
+    def __call__(self):
+        pass
+
+    def __str__(self):
+        from airflow.utils.db_cleanup import config_dict
+
+        return str(sorted(config_dict))
```

Comment:
The [original code](https://github.com/apache/airflow/blob/436c17c655494eff5724df98d1a231ffa2142253/airflow/cli/cli_parser.py#L421) used `lazy_object_proxy`:

```python
ARG_DB_TABLES = Arg(
    ("-t", "--tables"),
    help=lazy_object_proxy.Proxy(
        lambda: f"Table names to perform maintenance on (use comma-separated list).\n"
        f"Options: {import_string('airflow.cli.commands.db_command.all_tables')}"
    ),
    type=string_list_type,
)
```

What do you think of using that instead?

Reply:
```diff
commit a71ef4a4372087e46cae970d8a7c45c59f790f9e (HEAD -> fix-imports-use-lazy-object-proxy, origin/fix-imports-use-lazy-object-proxy)
Author: blag <blag@users.noreply.github.com>
Date:   Mon Apr 4 14:00:27 2022 -0700

    Use lazy_object_proxy.Proxy

diff --git a/airflow/cli/commands/db.py b/airflow/cli/commands/db.py
index bf7357d33..b0d41bb15 100644
--- a/airflow/cli/commands/db.py
+++ b/airflow/cli/commands/db.py
@@ -19,6 +19,7 @@ import os
 import textwrap
 from tempfile import NamedTemporaryFile

+import lazy_object_proxy
 import rich_click as click
 import wrapt
 from packaging.version import parse as parse_version
@@ -302,16 +303,13 @@ def check(ctx, migration_wait_timeout):


 # lazily imported by CLI parser for `help` command
-# Create a custom class that emulates a callable since click validates
-# non-callable and make __str__ to return output for lazy processing.
-class _CleanTableDefault:
-    def __call__(self):
-        pass
+def _get_all_tables():
+    from airflow.utils.db_cleanup import config_dict

-    def __str__(self):
-        from airflow.utils.db_cleanup import config_dict
+    return sorted(config_dict)

-        return str(sorted(config_dict))
+
+all_tables = lazy_object_proxy.Proxy(_get_all_tables)


 @db.command('cleanup')
@@ -320,7 +318,7 @@ class _CleanTableDefault:
     '-t',
     '--tables',
     multiple=True,
-    default=_CleanTableDefault(),
+    default=all_tables,
     show_default=True,
     help=(
         "Table names to perform maintenance on (use comma-separated list).\n"
```

It's not super clean, but it is slightly cleaner than the class-based version.

Reply:
The patch still seems to be import heavy as per my tests. Can you please confirm timings on your machine after this change?

```
With class : 
root@e39a683328ff:/opt/airflow# python -X importtime -m airflow.cli 2>&1 | wc 
    655    4548   45806

With lazy_object_proxy

root@e39a683328ff:/opt/airflow# python -X importtime -m airflow.cli 2>&1 | wc 
   1330    9443  100432
```

Reply:
Good call, you're right. :+1:

```bash
python -X importtime -m airflow.cli 2>&1 | wc  # with class
     657    4667   46511

python -X importtime -m airflow.cli 2>&1 | wc  # lazy_object_proxy
     969    7021   71989
```